### PR TITLE
Custom attribute mappings

### PR DESCRIPTION
**Pull request by juliushaertl**
- Allow mapping of uid/displayName/email/quota
- Add option to turn off the unique hasing of user ids
- Allow updating existing provider configurations
- Extend frontend for new settings and provider updating

based on #267 
implements #42

## Updated settings UI

image
image

### CODE REVIEW

File: lib/Settings/AdminSettings.php [hunk 1]
```diff
@@ -37,7 +38,7 @@
 class AdminSettings implements ISettings {
 
 	/** @var ProviderMapper */
```
Comment (blizzz) [inline lib/Settings/AdminSettings.php:40]: 😕

File: package.json [hunk 2]
```diff
@@ -24,6 +24,7 @@
 		"@nextcloud/initial-state": "^1.2.0",
 		"@nextcloud/logger": "^2.0.0",
 		"@nextcloud/router": "^2.0.0",
+		"@nextcloud/vue": "^4.0.0-alpha.4",
```
Comment (blizzz) [inline package.json:27]: alpha?

**Review by blizzz — COMMENTED** [review]
Looks good, test might be a bit shallow. Comments inline.

**Review by blizzz — APPROVED** [review]
didn't test, looks good

**Comment by juliushaertl** [comment]
Put some basic unit tests for the new logic except for the controller which should rather be covered by proper integration tests anyways. First attempts to fire up some presetup oidc provider didn't work well but I'll keep looking into that in a separate pr then.

**Pull request merged**